### PR TITLE
Message request fixes and tweaks

### PR DESCRIPTION
**Pull request by mpretty-cyro**
- Removed the "Back" text from the back buttons
- Removed the inset on the 'Path' settings button so the text remains horizontally centered
- Hid the settings button from message request threads
- Fixed an issue where the back button would remain visible in a conversation when the search field was visible
- Fixed an issue where the tintColor of the conversation search field didn't match the global search field
- Fixed an issue where sending an attachment response to a message request wouldn't approve the message request
- Updated the size and positioning of the message request 'Clear All' button to match the DM 'Next' button
- Updated the message request 'Clear All' button to start visible (so it's visible during the push animation) since that's the most likely state it'll be in
- Updated the 'Message Requests' cell to use the pinned background colour
- Updated the fallback for contact thread names to be a middle-truncated string (4 characters either side)

### CODE REVIEW

File: Session/Conversations/ConversationVC.swift [hunk 1]
```diff
@@ -465,30 +466,53 @@ final class ConversationVC : BaseVC, ConversationViewModelDelegate, OWSConversat
...
+        
+        navigationItem.hidesBackButton = isShowingSearchUI
+        
         if isShowingSearchUI {
             navigationItem.leftBarButtonItem = nil
             navigationItem.rightBarButtonItems = []
-        } else {
+        }
+        else {
+            navigationItem.hidesBackButton = false
```
Comment (RyanRory) [inline Session/Conversations/ConversationVC.swift:481]: This line seems to be redundant to `navigationItem.hidesBackButton = isShowingSearchUI`

File: Session/Conversations/ConversationVC.swift [hunk 2]
```diff
@@ -465,30 +466,53 @@ final class ConversationVC : BaseVC, ConversationViewModelDelegate, OWSConversat
     // MARK: Updating
     
     func updateNavBarButtons() {
+        // Back button (to appear on pushed screen)
+        let backButton = UIBarButtonItem(title: "", style: .plain, target: nil, action: nil)
+        backButton.tintColor = Colors.text
+        navigationItem.backBarButtonItem = backButton
```
Comment (RyanRory) [inline Session/Conversations/ConversationVC.swift:472]: Can we do this in `BaseVC` so we can remove all back button text of the VCs inheriting `BaseVC`?

File: Session/Home/HomeVC.swift [hunk 3]
```diff
@@ -393,11 +393,18 @@ final class HomeVC : BaseVC, UITableViewDataSource, UITableViewDelegate, NewConv
         profilePictureViewContainer.addSubview(pathStatusView)
         pathStatusView.pin(.trailing, to: .trailing, of: profilePictureViewContainer)
         pathStatusView.pin(.bottom, to: .bottom, of: profilePictureViewContainer)
+        
+        // Back button (to appear on pushed screen)
+        let backButton = UIBarButtonItem(title: "", style: .plain, target: nil, action: nil)
+        backButton.tintColor = Colors.text
+        navigationItem.backBarButtonItem = backButton
```
Comment (RyanRory) [inline Session/Home/HomeVC.swift:400]: Same as the back button in `ConversationVC`. :p

File: Session/Settings/SettingsVC.swift [hunk 4]
```diff
@@ -139,7 +139,7 @@ final class SettingsVC : BaseVC, AvatarViewHelperDelegate {
         setUpNavBarStyle()
         setNavBarTitle(NSLocalizedString("vc_settings_title", comment: ""))
         // Navigation bar buttons
-        let backButton = UIBarButtonItem(title: "Back", style: .plain, target: nil, action: nil)
+        let backButton = UIBarButtonItem(title: "", style: .plain, target: nil, action: nil)
```
Comment (RyanRory) [inline Session/Settings/SettingsVC.swift:142]: Same as the back button in `HomeVC` & `ConversationVC` :)

File: SignalUtilitiesKit/Utilities/UIViewController+Utilities.swift [hunk 5]
```diff
@@ -28,7 +28,7 @@ public final class ViewControllerUtilities : NSObject {
         }
         // Set up back button
         if hasCustomBackButton {
-            let backButton = UIBarButtonItem(title: "Back", style: .plain, target: nil, action: nil)
+            let backButton = UIBarButtonItem(title: "", style: .plain, target: nil, action: nil)
```
Comment (RyanRory) [inline SignalUtilitiesKit/Utilities/UIViewController+Utilities.swift:31]: This is interesting. Seems like we really have a lot of redundant code. Could you clean up a bit on all these code related to back button? Cheers!

File: Session/Conversations/ConversationVC.swift (hunk 1 above)
Comment (mpretty-cyro) [reply Session/Conversations/ConversationVC.swift:481]: Ah good catch, will remove

File: SignalUtilitiesKit/Utilities/UIViewController+Utilities.swift (hunk 5 above)
Comment (mpretty-cyro) [reply SignalUtilitiesKit/Utilities/UIViewController+Utilities.swift:31]: Yea this is a bit of a messy one - I think the main difference is when view controllers don't inherit from `BaseVC` they use this method (ie. cases where things inherit from `OWSTableViewController`/`OWSViewController`/`UICollectionViewController`) - seems to be used in ~7 places and I'm not sure we want to refactor those screens to use `BaseVC` as part of this PR

Comment (RyanRory) [reply SignalUtilitiesKit/Utilities/UIViewController+Utilities.swift:31]: OK, no worries! Let's leave this cleaning up for future us!

**Pull request merged**